src/Quartz/Logging/IJobDiagnosticData.cs

```diff
@@ -0,0 +1,138 @@
+using System;
+
+namespace Quartz.Logging
+{
+    public interface IJobDiagnosticData
```

Comment:
This interface is public because I'm going to use it in my project. I enlighted it in #1966
I made an interface not because there's going to be other implementations but because it is a contract, abstraction like a `IJobExecutionContext` and I followed dependency inversion principle.
If you'd like I'll delete `IJobDiagnosticData` and will make `JobDiagnosticData` `public`.